plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/CachingJdbcClient.java

```diff
@@ -344,7 +344,13 @@ public PreparedStatement getPreparedStatement(Connection connection, String sql)
     @Override
     public TableStatistics getTableStatistics(ConnectorSession session, JdbcTableHandle handle, TupleDomain<ColumnHandle> tupleDomain)
     {
-        TableStatisticsCacheKey key = new TableStatisticsCacheKey(handle, tupleDomain);
+        return getTableStatistics(session, handle);
+    }
+
+    @Override
+    public TableStatistics getTableStatistics(ConnectorSession session, JdbcTableHandle handle)
+    {
+        TableStatisticsCacheKey key = new TableStatisticsCacheKey(handle);
```

Comment:
nit: `TableStatisticsCacheKey` is no longer needed, can be replaced with `JdbcTableHandle`, provided that we retain
```
// TODO depend on Identity when needed
```

comment eg here, within the getTableStatistic() method

(optional)

Reply:
Added in a separate commit.